### PR TITLE
fix: do not ask to allow incoming connections on macOS

### PR DESCRIPTION
**Pull request by matt2e**
macOS prompts for permission if we do not specify `127.0.0.1`

### CODE REVIEW

File: internal/bind/bind_allocator.go [hunk 1]
```diff
@@ -35,7 +35,11 @@ func (b *BindAllocator) Next() *url.URL {
 	var err error
 	for {
 		b.port.Add(1)
-		l, err = net.ListenTCP("tcp", &net.TCPAddr{IP: net.ParseIP(b.baseURL.Hostname()), Port: int(b.port.Load())})
+		host := b.baseURL.Hostname()
```
Comment (alecthomas) [inline internal/bind/bind_allocator.go:38]: Let's just change the default addresses to `127.0.0.1` then we don't need to special case this here.

**Review by alecthomas — APPROVED** [review]

**Comment by alecthomas** [comment]
I think there are a few more locations where `default:"http://localhost` is used

**Comment by matt2e** [comment]
@alecthomas ah, updated the runner defaults as welll

**Comment by worstell** [comment]
oh my god thank you

**Comment by gak** [comment]
thanks! 🥇

**Pull request merged**